docs/datastore/crud-operations.md

```diff
@@ -0,0 +1,58 @@
+---
+id: crud-operations
+title: CRUD Operations
+sidebar_label: CRUD Operations
+---
+
+All interactions are done with the on-device storage.
+
+## Saving data
+
+```typescript
+import { TaskModel } from './datastoreConfig';
+
+TaskModel.save({
+    title: "Write Docs",
+    description: "Write Offix Docs",
+}).then((data) => {
+    console.log(data); // { id: '...', title, description }
+})
+```
+
+## Querying data
+
+```typescript
+import { TaskModel } from './datastoreConfig';
+
+TaskModel.query().then((data) => {}) // Retrieves all tasks
+```
+
+You can filter data using the schema fields defined in the model
+
+```typescript
+TaskModel.query({ title: "test" })
+.then((data) => {}) // Retrieves all tasks where title matches "test"
+```
+
+## Updating data
+
+```typescript
+import { TaskModel } from './datastoreConfig';
+
+// update the title of all tasks with title = "test" to "Offix Test"
```

Comment:
This case is rare! We should show update by id. IMHO updating multiple items in client should not be even possible (it brings serious risk and makes it hard to implement security)

Reply:
This doc is outdated. It hasn't been updated to include new API that has "ById" methods.  Also, why does this make it hard to implement security? Security is done on the server and we don't push batch updates to server.